src/lib/error.rs

```diff
@@ -57,3 +57,12 @@ impl std::convert::From<std::io::Error> for NisporError {
         }
     }
 }
+
+impl std::convert::From<&str> for NisporError {
```

Comment:
Converting `&str` to `NisporError` seems too dangerous to me, normally we only convert other type of errors into NisporError.

You may try to implement the `NisporError::bug(msg: &str)` to simply the effort on create a `NisporError` with `ErrorKind::NisporBug`

Reply:
Done! I've implemented the bug() function for NisporError. Thanks!